processor/src/main/java/org/mapstruct/ap/internal/util/workarounds/SpecificCompilerWorkarounds.java

```diff
@@ -149,4 +156,45 @@ static TypeMirror asMemberOf(Types typeUtils, ProcessingEnvironment env, Version
 
         return result;
     }
+
+    /**
+     * In Eclipse 4.6, the ClassLoader of the annotation processor does not provide access to the implementation types
+     * of the APT-API anymore, so we need to create a new ClassLoader containing the processor class path URLs and
+     * having the ClassLoader of the APT-API implementations as parent. The method invocation then consequently needs to
+     * be done via reflection.
+     *
+     * @author Andreas Gudian
+     */
+    private static final class ClassLoaderHelper {
+        private static ClassLoader classLoader;
+        private static Method asMemberOf;
+
+        private static ClassLoader getOrCreateClassLoader(ClassLoader parent) {
+            if ( classLoader == null ) {
+                classLoader = new URLClassLoader(
+                    ( (URLClassLoader) ClassLoaderHelper.class.getClassLoader() ).getURLs(), parent );
+            }
+
+            return classLoader;
+        }
+
+        private static Method getAsMemberOf(ClassLoader platformClassLoader) throws Exception {
+            if ( asMemberOf == null ) {
+                Class<?> workaroundClass = getOrCreateClassLoader( platformClassLoader )
+                    .loadClass( ECLIPSE_AS_MEMBER_OF_WORKAROUND );
```

Comment:
Ok.. But.. you could also choose to wrap this part in a new `EclipseSpecificCompilerWorkaround` and delegate to `EclipseAsMemberOfWorkaround`. Advantage: the class loader stuff is Eclipse specific, so it would make sense to isolate that.